server/src/main/java/io/crate/planner/operators/Get.java

```diff
@@ -135,20 +147,55 @@ public ExecutionPlan build(PlannerContext plannerContext,
                 .orElse(SequenceNumbers.UNASSIGNED_PRIMARY_TERM);
             pkAndVersions.add(new PKAndVersion(id, version, sequenceNumber, primaryTerm));
         }
-        return new Collect(
+
+        var docKeyColumns = new ArrayList<>(docTableInfo.primaryKey());
+        docKeyColumns.addAll(docTableInfo.partitionedBy());
+        docKeyColumns.add(docTableInfo.clusteredBy());
+        docKeyColumns.add(DocSysColumns.VERSION);
+        docKeyColumns.add(DocSysColumns.SEQ_NO);
+        docKeyColumns.add(DocSysColumns.PRIMARY_TERM);
+
+        List<Symbol> boundOutputs = Lists2.map(
+            outputs, s -> SubQueryAndParamBinder.convert(s, params, subQueryResults));
+        var boundQuery = SubQueryAndParamBinder.convert(query, params, subQueryResults);
```

Comment:
```suggestion
        var binder = new SubQueryAndParamBinder(params, subQueryResults);
        List<Symbol> boundOutputs = Lists2.map(outputs, binder)
        var boundQuery = binder.convert(query);
```

Avoids creating 2 SubQueryAndParamBinder instances